Libplanet/Blocks/PreEvaluationBlockHeader.cs

```diff
@@ -148,44 +146,27 @@ public PreEvaluationBlockHeader(IBlockMetadata metadata)
                         $"the previous block's hash {metadata.PreviousHash}.");
                 }
 
-                if (commit.Votes is { } votes)
+                if (commit.Votes is { } votes && !votes.All(vote => vote.Height == commit.Height))
```

Comment:
This can be checked internally from `BlockCommit`. I think its better to have this as part of `BlockCommitExtensions`.

Reply:
Amended.
https://github.com/planetarium/libplanet/blob/15f67c376b0bade3c0a32b0ed3475d42fd44f160/Libplanet/Blocks/BlockCommitExtensions.cs#L67-L74